src/overmap_location.cpp

```diff
@@ -65,3 +71,21 @@ void overmap_locations::reset()
 {
     locations.reset();
 }
+
+const std::vector<overmap_location> &overmap_locations::get_all()
+{
+    return locations.get_all();
+}
+
+const std::string overmap_locations::get_random_terrain( const std::string target_location_id )
+{
+    std::string random_terrain;
+    const std::vector<overmap_location> locs = overmap_locations::get_all();
+    for( const auto &loc : locs ) {
+        if( target_location_id == loc.id.str() ) {
```

Comment:
You don't want linear search here, the identifier (if it's not a `string`) is sufficient.